subprojects/core-api/src/main/java/org/gradle/api/resources/TextResourceFactory.java

```diff
@@ -85,4 +89,15 @@
      * Same as {@code fromArchiveEntry(archive, path, Charset.defaultCharset().name())}.
      */
     TextResource fromArchiveEntry(Object archive, String path);
+
+    /**
+     * Creates a text resource backed by the given uri.
+     *
+     * @param uri an uri
+     *
+     * @return a text resource backed by the given uri
+     * @since TODO
```

Comment:
I was not sure which version should go here, I suppose `4.2`. Once you let me know, I can update it without problem